hardware/battery/battery_management_system_protocol.ksy

```diff
@@ -0,0 +1,223 @@
+meta:
+  id: battery_management_system_protocol
+  title: Communication protocol of smart battery management systems from LLT power
+  license: CC0-1.0
+  ks-version: 0.8
+  endian: be
+
+doc: |
+  Many modern general purpose BMS include a UART/Bluetooth based communication interface.
+  After sending read requests they respond with various information's about the battery state in
+  a custom binary format.
+
+doc-ref: https://www.lithiumbatterypcb.com/Protocol%20English%20Version.rar
+
+seq:
+  - id: magic_start
+    contents: [0xdd]
+  - id: cmd
+    type: u1
+
+  - id: body
+    type:
+      switch-on: cmd
+      cases:
+        0xa5: read_req
+        0x5a: write_req
+        _: response(cmd)
+  
+  - id: checksum
+    size: 2
+  - id: magic_end
+    contents: [0x77]
+
+types:
+  read_req:
+    seq:
+      - id: req_cmd
+        type: u1
+        doc: Same value as cmd for response
+      - id: data_len
+        contents: [0x00]
+  write_req:
+    seq:
+      - id: req_cmd
+        type: u1
+        doc: Same value as cmd for response
+      - id: data_len
+        type: u1
+      - id: write_data
+        size: data_len
+
+  basic_info:
+    seq:
+      - id: total
+        type: voltage
+      - id: current
+        type: current
+      - id: remain_cap
+        type: capacity
+      - id: typ_cap
+        type: capacity
+      - id: cycles
+        type: u2
+        doc: Cycle times
+      - id: prod_date
+        type: u2
+        doc: Production date
+      - id: balance_status
+        type: balance_list
+        doc: List of balance bits
+      - id: prot_status
+        type: prot_list
+        size: 2
```

Comment:
seems to be unneeded

Reply:
True, size is redundant, it is removed now 042fc91